src/fsharp/fsi/fsi.fs

```diff
@@ -2755,7 +2755,14 @@ type FsiEvaluationSession (fsi: FsiEvaluationSessionHostConfig, argv:string[], i
     let currentDirectory = Directory.GetCurrentDirectory()
     let tryGetMetadataSnapshot = (fun _ -> None)
 
-    let defaultFSharpBinariesDir = FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(FSharpEnvironment.tryCurrentDomain()).Value
+    let defaultFSharpBinariesDir = 
+        try
+            FSharpEnvironment.BinFolderOfDefaultFSharpCompiler(FSharpEnvironment.tryCurrentDomain()).Value
+        with
+        | _ ->
+            let d = System.Reflection.Assembly.GetExecutingAssembly() //IO.Directory.GetCurrentDirectory()        
```

Comment:
Why not use `AppContext.BaseDirectory`?

Reply:
When using Python.net or native .net core hosting scenario to load DLR libraries, it seems AppContext.BaseDirectory doesn't work... (include configuration manager, app.config...)

Reply:
https://docs.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting